plugins/services/src/js/components/ServiceBreadcrumbs.js

```diff
@@ -45,6 +44,22 @@ class ServiceBreadcrumbs extends React.Component {
     global.addEventListener("resize", this.handleViewportResize);
   }
 
+  shouldComponentUpdate(nextProps) {
+    const hasServiceIDChanged = this.props.serviceID !== nextProps.serviceID;
+    const hasTaskIDChanged = this.props.taskID !== nextProps.taskID;
+    const hasTaskNameChanged = this.props.taskName !== nextProps.taskName;
+    const hasExtraNotChanged =
```

Comment:
Why is this the only time where it is not changed instead of changed?

Reply:
Good question. The extra prop is an array of nodes. But they are only set on certain occasions so in most cases, extra will be undefined. This is caught by the first comparison. If checking for changes the comparison would be a lot more complex that is why I picked the not changed instead. 

Just to give more meaning to that the comparison would start with something like that:
```JS
(this.props.extra === undefined && nextProps.extra !== undefined) ||
(this.props.extra !== undefined && nextProps.extra === undefined) ||
!deepEqual(this.props.extra, nextProps.extra)
```

And I would also need to make sure to check for null. So I choose to rather have one Not Changed instead of a Changed. But also I could wrap the condition into brackets so it would look like this:
```JS
const hasExtraChanged = !( 
  his.props.extra === nextProps.extra ||
  deepEqual(this.props.extra, nextProps.extra)
);
```
But my Opinion is that this more complex.

Reply:
I think this looks readable:

```js
const hasExtraChanged = this.props.extra !== nextProps.extra || !deepEqual(this.props.extra, nextProps.extra);
```

Reply:
Offline discussed: The first part is truthy if it's a different object which is completely equal